### ISSUE TITLE
"no-mixed-operators" always outputs errors in pairs of two

### ISSUE THREAD

**Issue by feross** [eslint/eslint#8051]
**Tell us about your environment**

* **ESLint Version:** 3.15.0
* **Node Version:** 7.5.0
* **npm Version:** 4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```
    "no-mixed-operators": ["error", {
        "groups": [
            ["+", "-", "*", "/", "%", "**"],
            ["==", "!=", "===", "!==", ">", ">=", "<", "<="],
            ["&&", "||"],
            ["in", "instanceof"]
        ],
        "allowSamePrecedence": true
    }],
```

**What did you do? Please include the actual source code causing the issue.**

```js
var eLen = nBytes * 8 - mLen - 1
```

**What did you expect to happen?**

One error about mixed operators.

**What actually happened? Please include the actual, raw output from ESLint.**

Two errors about mixed operators, one for each operator:

```
  index.js:3:21: Unexpected mix of '*' and '-'. (no-mixed-operators)
  index.js:3:25: Unexpected mix of '*' and '-'. (no-mixed-operators)
```

Is this expected behavior? It seems there is no case where this rule emits a single error, since by definition "mixing" operators requires 2 operators. It would be more intuitive to me if a single error was emitted.

**Comment by not-an-aardvark** [comment]
This seems like a bug. Thanks for the report!

**Comment by not-an-aardvark** [comment]
This seems like a bug. Thanks for the report!

**Comment by mysticatea** [comment]
Actually, this is intentional.

`Unexpected mix of '*' and '-'.`, in this case, it shows 2 warnings at the `*` and `-` to make clear what operators are mixed.

**Comment by feross** [comment]
@mysticatea Wouldn't showing a single warning at the first operator suffice?

**Comment by mysticatea** [comment]
I will not oppose it. However, this rule warns consecutive 2 operators on AST, so it might be at far locations on source code text if the expression is complex. It was the reason that I made 2 warnings.

**Comment by feross** [comment]
Makes sense. Either way is fine, though I was initially confused by the double error.

**Comment by nzakas** [comment]
I think this is the correct behavior as well. Perhaps we should update the docs to mention this?
